akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSinkSpec.scala

```diff
@@ -109,7 +110,7 @@ class QueueSinkSpec extends StreamSpec {
       sub.sendComplete()
       Await.result(queue.pull(), noMsgTimeout) should be(None)
 
-      queue.pull().failed.foreach { e ⇒ e.isInstanceOf[IllegalStateException] should ===(true) }
+      queue.pull().failed.futureValue shouldBe an[AbruptStageTerminationException]
```

Comment:
Let me introduce new specific exception as it will be reused with several sink/sources.
(I don't like idea of using Abrupt exception either as it seems too internal)
Yes, I'm on it